Foundation/FileManager.swift

```diff
@@ -442,16 +442,34 @@ open class FileManager : NSObject {
             else {
                 return
         }
+
+        func copyNonDirectory(srcPath: String, dstPath: String, fileType: FileAttributeType) throws {
+            if fileType == .typeSymbolicLink {
+                let destination = try destinationOfSymbolicLink(atPath: srcPath)
+                try createSymbolicLink(atPath: dstPath, withDestinationPath: destination)
+            } else if fileType == .typeRegular {
+                if createFile(atPath: dstPath, contents: contents(atPath: srcPath), attributes: nil) == false {
```

Comment:
This works, but a proper copy engine will read then write a limited portion of the source file at a time. For large files, this will result in an extreme high-water mark for memory usage.